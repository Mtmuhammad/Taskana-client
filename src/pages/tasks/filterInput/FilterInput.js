// text input to search and filter tasks

const FilterInput = ({ handleChange }) => {
  return (
    <div className="p-3 border-bottom">
      <div data-testid="input-container" className="input-group searchbar">
        <span
          data-testid="input-icon"
          className="input-group-text text-muted"
          id="search"
        >
          <i className="bx bx-search-alt-2"></i>
        </span>
        <input
          data-testid="input-search"
          type="text"
          className="form-control"
          placeholder="Search Tasks Here"
          id="task-search"
          onChange={handleChange}
        />
      </div>
    </div>
  );
};

export default FilterInput;
