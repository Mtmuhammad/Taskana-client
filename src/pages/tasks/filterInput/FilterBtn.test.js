import React from "react";
import { render, screen } from "@testing-library/react";
import FilterInput from "./FilterInput";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<FilterInput />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<FilterInput />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("filter input", () => {
  test("should render filter input container", () => {
    render(<FilterInput />, { wrapper: BrowserRouter });
    const container = screen.getByTestId("input-container");
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("input-group searchbar");
  });
  test("should render filter input icon", () => {
    render(<FilterInput />, { wrapper: BrowserRouter });
    const icon = screen.getByTestId("input-icon");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("input-group-text text-muted");
  });
  test("should render filter input", () => {
    render(<FilterInput />, { wrapper: BrowserRouter });
    const input = screen.getByTestId("input-search");
    expect(input).toBeInTheDocument();
    expect(input).toHaveClass("form-control");
  });
});
