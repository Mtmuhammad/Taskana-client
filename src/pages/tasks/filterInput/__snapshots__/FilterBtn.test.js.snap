// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-3 border-bottom"
  >
    <div
      class="input-group searchbar"
      data-testid="input-container"
    >
      <span
        class="input-group-text text-muted"
        data-testid="input-icon"
        id="search"
      >
        <i
          class="bx bx-search-alt-2"
        />
      </span>
      <input
        class="form-control"
        data-testid="input-search"
        id="task-search"
        placeholder="Search Tasks Here"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
